docs/dependencies.md

```diff
@@ -0,0 +1,287 @@
+  `scripts/lib/setup_path_on_import.py`, which on import will put the
+  currently running Python script into the Zulip virtualenv.  This is
+  called by `./manage.py` to ensure that our Django code always uses
+  the correct virtualenv as well.
+
+## JavaScript and other frontend packages
+
+We use the same set of strategies described for Python dependencies
+for most of our JavaScript dependencies.  Consult the above section
+for details on the strategy; here we discuss how things differ.
+
+* **node_modules directory with pinned versions**.  We store
+  `node_modules` directories under `/srv/zulip-npm-cache` in a fashion
+  analogous to the virtualenvs for Python packages.
+* We use [yarn][], which talks to the [npm][] repository, to download
+  most JavaScript dependencies.  We use the standard `package.json`
+  file to declare our direct dependencies, both for development and
+  production.  Yarn takes care of pinning the versions of indirect
+  dependencies in the `yarn.lock` file.
+* **Caching**.  We use `scripts/lib/node_cache.py` for managing our
+  cached `node_modules` directories containing copies of our
+  depedenencies.
+* **Garbage-collecting caches**.  `scripts/lib/clean-npm-cache` is
+  very similar to `clean-venv-cache`.
+* `tools/update-prod-static`.  This process is discussed in detail in
+  the [static asset pipeline](/front-end-build-process.html) article,
+  but we don't use the `node_modules` directories directly in
+  production.  Instead, static assets are compiled using our static
+  asset pipeline and it is the compiled assets that are served
+  directly to users.  As a result, we don't ship the `node_modules`
+  directory in a Zulip production release tarball, which is a good
+  thing, because doing so would more than double the size of a Zulip
+  release tarball.
+
+* **Checked-in packages** In contrast with Python, we have a few
+JavaScript depedencies that we have copied into the main Zulip
+repository under `static/third`, often with patches.  These date from
+an era before `npm` existed.  It is a project goal to eliminate these
+checked-in versions of dependencies and instead use versions managed
+by the npm repositories.
+
+## Node and Yarn
+
+These are installed by `scripts/lib/install-node` (which in turn uses
+the standard third-party `nvm` installer to download `node` and pin
+its version) and `scripts/lib/third/install-yarn.sh` (the standard
+installer for `yarn`, modified to support installing to a path that is
+not the current user's home directory).
+
+* `nvm` has its own system for installing each version of `node` at
+its own path, which we use, though we install a `/usr/local/bin/node`
+wrapper to access the desired version conveniently and efficiently
+(`nvm` has a lot of startup overhead).
+* `install-yarn.sh` is configured to install `yarn` at
+`/srv/zulip-yarn`.  We don't do anything special to try to manage
+multiple versions of `yarn`.
+
+## Emoji data and other generated files
```

Comment:
Maybe "Other third-party and generated files"? That'd better match the description in the next sentence. And "Emoji" is the first sub-heading, which seems good enough for finding the discussion of emoji data.